src/preferences/preference-service.ts

```diff
@@ -0,0 +1,52 @@
+import { Emitter, Event } from "../application/common";
+
+
+export class PreferenceService {
+    protected readonly onPreferenceChangedEmitter = new Emitter<PreferenceChangeEvent>();
+
+    preferences: Map<string, Preference> = new Map<string, Preference>();
+
+    protected firePreferenceChanged(pref: PrefChange) {
+        this.onPreferenceChangedEmitter.fire(new PreferenceChangeEvent(pref));
+    }
+
+    get onPreferenceChange(): Event<any> {
+        return this.onPreferenceChangedEmitter.event;
+    }
+
+    /**
+     * Get preference by key if it exists, return null if not
+     * @param key Unique preference string identifier
+     */
+    getPref(key: string) {
+
+        let pref: Preference | undefined = this.preferences.get(key);
+        return pref !== undefined ? pref : null;
+    }
+
+    /**
+     * Updates a pref and fires an appropriate event for it
+     * @param pref The new preference
+     */
+    setPref(pref: Preference) {
```

Comment:
I don't think we should expose setters for now, at least not from the same interface.
Currently I assume all changes going through the file system